hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rollout/RolloutScheduler.java

```diff
@@ -44,12 +48,29 @@ public RolloutScheduler(
     public void runningRolloutScheduler() {
         log.debug("rollout schedule checker has been triggered.");
 
-        // run this code in system code privileged to have the necessary permission to query and create entities.
+        // run this code in system code privileged to have the necessary
+        // permission to query and create entities.
         systemSecurityContext.runAsSystem(() -> {
-            // workaround eclipselink that is currently not possible to execute a query without multi-tenancy if MultiTenant
-            // annotation is used. https://bugs.eclipse.org/bugs/show_bug.cgi?id=355458. So, iterate through all tenants and execute the rollout
-            // check for each tenant separately.
-            systemManagement.forEachTenant(tenant -> rolloutHandler.handleAll());
+            // workaround eclipselink that is currently not possible to
+            // execute a query without multi-tenancy if MultiTenant
+            // annotation is used.
+            // https://bugs.eclipse.org/bugs/show_bug.cgi?id=355458. So
+            // iterate through all tenants and execute the rollout check for
+            // each tenant seperately.
+
+            systemManagement.forEachTenant(tenant ->
+                rolloutTaskExecutor.submit(() ->
+                    systemSecurityContext.runAsSystemAsTenant(() -> {
+                        try {
+                            log.trace("Handling rollout for tenant: {}", tenant);
```

Comment:
handleAll actually logs db errors and caches (most of) the exceptions ..
but this logging could handle additionally locking exceptions.
However, for consistency - maybe it's good to put all the logging?
Not sure, though. It could be seen as different scheduler handling ...
Just for consideration